code/game/turfs/turf.dm

```diff
@@ -318,6 +318,30 @@ GLOBAL_LIST_EMPTY(station_turfs)
 
 	return FALSE
 
+/turf/attackby_secondary(obj/item/weapon, mob/user, params)
+	if(istype(weapon, /obj/item/stack/sheet/iron))
+		var/obj/item/stack/sheet/iron/using_sheet = weapon
+		if(!isopenturf(src) || isgroundlessturf(src))
+			user.balloon_alert(user, "can't place it here.")
+			return SECONDARY_ATTACK_CONTINUE_CHAIN
```

Comment:
All of these should probably cancel instead of continuing the chain.

Reply:
Done